users/service.go

```diff
@@ -45,21 +71,39 @@ type Service interface {
 	// other reason, non-nil error values are returned in response.
 	Identify(string) (string, error)
 
-	// Get authenticated user info for the given token.
+	// Get authenticated user info for the given token
 	UserInfo(ctx context.Context, token string) (User, error)
+
+	// GenerateResetToken email where mail will be sent.
+	// host is used for generating reset link.
+	GenerateResetToken(_ context.Context, email, host string) error
+
+	// UpdatePassword change users password in reset flow
+	UpdatePassword(_ context.Context, token, password string) error
+
+	//SendToken sends reset password link to email
+	SendToken(_ context.Context, host, email, token string) error
 }
 
 var _ Service = (*usersService)(nil)
 
+// Emailer used for sending messages to user
+type Emailer struct {
+	ResetURL string
+	Agent    *email.Agent
+}
```

Comment:
`Emailer` can be in the `email` package, and here you can create `Emailer` interface with `Send` method.

Reply:
Idea is that email package can be reusable in other services ResetURL is users specific so that is why I put it here. Same comment goes for the message text.